### PR TITLE
reason and description fields are added to refund and cancel requests

### PR DESCRIPTION
**Pull request by ktoprakucar**

### CODE REVIEW

File: src/Iyzipay/Request/CreateRefundRequest.php [hunk 1]
```diff
@@ -12,6 +12,8 @@ class CreateRefundRequest extends Request
     private $price;
     private $ip;
     private $currency;
+    private $reason;
```
Comment (kadriyedogan) [inline src/Iyzipay/Request/CreateRefundRequest.php]: We can add reason enum values like Locale,Currency, etc.

File: samples/cancel_with_reason.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+require_once('config.php');
+
+# create request class
+$request = new \Iyzipay\Request\CreateCancelRequest();
+$request->setLocale(\Iyzipay\Model\Locale::TR);
+$request->setConversationId("123456789");
+$request->setPaymentId("2");
```
Comment (sotuzun) [inline samples/cancel_with_reason.php]: PaymentId must be 1

File: samples/refund_with_reason.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+require_once('config.php');
+
+# create request class
+$request = new \Iyzipay\Request\CreateRefundRequest();
+$request->setLocale(\Iyzipay\Model\Locale::TR);
+$request->setConversationId("123456789");
+$request->setPaymentTransactionId("2");
```
Comment (sotuzun) [inline samples/refund_with_reason.php]: We need to ensure that all sample data must be same.

File: src/Iyzipay/Request/CreateRefundRequest.php (hunk 1 above)
Comment (sotuzun) [reply src/Iyzipay/Request/CreateRefundRequest.php]: You can't do it in PHP. Language has type inference feature.

**Pull request merged**